### ISSUE TITLE
[BUG | WINE] Autocomplete with multi files plugin + file menu closing problem

### ISSUE THREAD

**Issue by Keplyx** [Timocop/BasicPawn#4]
* **The autocomplete does not seem to work on secondary files**

Here I am in my plugin's main file, autocomplete is working fine
![selection_012](https://cloud.githubusercontent.com/assets/23726131/25348114/74ee3308-291d-11e7-91a6-aedc133d4cc0.png)

But here, I'm in a secondary file, included in the main file, and autocomplete is missing.
![selection_013](https://cloud.githubusercontent.com/assets/23726131/25348115/761a04c8-291d-11e7-9dc5-c4a7c6bd153e.png)

As you can see, the path to the include files is in the custom profile, which is active
![selection_014](https://cloud.githubusercontent.com/assets/23726131/25348196/b8febb30-291d-11e7-87dc-fb4b889182d3.png)

Do I need to do some extra steps to make it work or is it just because the include files are declared only in the main file?

* **File menu closing itself**

This is not a big deal, but when I hover my mouse over an item in the File menu, it closes itself after 1 second.
![selection_015](https://cloud.githubusercontent.com/assets/23726131/25348359/49b4e140-291e-11e7-9814-48e832cd0069.png)

Also, I was wondering, would it be possible to port the software natively to Linux (without using wine)? I saw some posts here and there talking about .net being able to be cross platform. This would be a big difference between Basic Pawn and SPEdit (which cannot even work properly under wine). I would love to help, but I don't know much about .net.

**Comment by Timocop** [comment]
> The autocomplete does not seem to work on secondary files

Can you post the a screenshot of the "Information" tab when working on menus.sp?

 > File menu closing itself

Thats actualy a Wine bug, i had that exact same behaviour on Ubuntu x86 with .NET 2.0. Not sure what causes it. Maybe some focus issue. Windows works fine.

Yes .NET can crossplatform, with [**Mono**](http://www.mono-project.com/).
I/You could use Mono but im not sure if the third party libaries survive that switch or Windows-only features.
Im using some Windows APIs not sure how to make that crossplatform friendly. (I have no clue about linux 😉)

**Comment by Keplyx** [comment]
> Can you post the a screenshot of the "Information" tab when working on menus.sp?

![selection_016](https://cloud.githubusercontent.com/assets/23726131/25352610/eba7fc36-292c-11e7-9284-683d0b550011.png)

> I/You could use Mono but im not sure if the third party libaries survive that switch or Windows-only features.
Im using some Windows APIs not sure how to make that crossplatform friendly. (I have no clue about linux :wink:)

I was just wondering, because it could fix issues created by wine, like the menus closing by themselves, and also improve the software's speed.

But as it work fine for now (no big problems) under wine, porting it is not very important.

**Comment by Timocop** [comment]
You dont mind sharing the chickenwars sources so i can investigate?

**Comment by Timocop** [comment]
Nvm found it xD

**Comment by Keplyx** [comment]
Yes it happens to every secondary files

**Comment by Timocop** [comment]
Im not sure tho, in your last Screenshot it uses a Unix path (/home/...) than a Windows path (C:\..) from Wine as  it should be.
Can you post the "application_error.log" in the BasicPawn main directory?

However, i get a different problem:
![grafik](https://cloud.githubusercontent.com/assets/22834512/25356137/7cc6c92a-2939-11e7-930f-fb790f966133.png)
Not sure what this is but seems like a TreeView problem which only happens on autocomplete updates (which fills the Object Browser TreeView)
Windows has none of those issues.

**Comment by Keplyx** [comment]
> Can you post the "application_error.log" in the BasicPawn main directory?

```
[23/04/2017 15:32:39]
ICSharpCode.TextEditor.Document.HighlightingDefinitionInvalidException: Could not load mode definition file '3a3ca976-81df-4e5f-8df0-ef4f3d977d4c.xshd'.
 ---> System.MissingMethodException: Method 'System.Drawing.Color.Grey' not found.
   at System.RuntimeType.InvokeMember(String name, BindingFlags bindingFlags, Binder binder, Object target, Object[] providedArgs, ParameterModifier[] modifiers, CultureInfo culture, String[] namedParams)
   at ICSharpCode.TextEditor.Document.HighlightColor..ctor(XmlElement el)
   at ICSharpCode.TextEditor.Document.HighlightBackground..ctor(XmlElement el)
   at ICSharpCode.TextEditor.Document.HighlightingDefinitionParser.Parse(DefaultHighlightingStrategy highlighter, SyntaxMode syntaxMode, XmlReader xmlReader)
   --- End of inner exception stack trace ---
   at ICSharpCode.TextEditor.Document.HighlightingDefinitionParser.Parse(DefaultHighlightingStrategy highlighter, SyntaxMode syntaxMode, XmlReader xmlReader)
   at ICSharpCode.TextEditor.Document.HighlightingManager.LoadDefinition(DictionaryEntry entry)
   at ICSharpCode.TextEditor.Document.HighlightingManager.FindHighlighter(String name)
   at ICSharpCode.TextEditor.TextEditorControl.SetHighlighting(String name)
   at BasicPawn.ClassSyntaxTools.UpdateTextEditorSyntax()
[23/04/2017 15:32:47]
ICSharpCode.TextEditor.Document.HighlightingDefinitionInvalidException: Could not load mode definition file '3a3ca976-81df-4e5f-8df0-ef4f3d977d4c.xshd'.
 ---> System.MissingMethodException: Method 'System.Drawing.Color.Grey' not found.
   at System.RuntimeType.InvokeMember(String name, BindingFlags bindingFlags, Binder binder, Object target, Object[] providedArgs, ParameterModifier[] modifiers, CultureInfo culture, String[] namedParams)
   at ICSharpCode.TextEditor.Document.HighlightColor..ctor(XmlElement el)
   at ICSharpCode.TextEditor.Document.HighlightBackground..ctor(XmlElement el)
   at ICSharpCode.TextEditor.Document.HighlightingDefinitionParser.Parse(DefaultHighlightingStrategy highlighter, SyntaxMode syntaxMode, XmlReader xmlReader)
   --- End of inner exception stack trace ---
   at ICSharpCode.TextEditor.Document.HighlightingDefinitionParser.Parse(DefaultHighlightingStrategy highlighter, SyntaxMode syntaxMode, XmlReader xmlReader)
   at ICSharpCode.TextEditor.Document.HighlightingManager.LoadDefinition(DictionaryEntry entry)
   at ICSharpCode.TextEditor.Document.HighlightingManager.FindHighlighter(String name)
   at ICSharpCode.TextEditor.TextEditorControl.SetHighlighting(String name)
   at BasicPawn.ClassSyntaxTools.UpdateTextEditorSyntax()
[23/04/2017 17:17:16]
ICSharpCode.TextEditor.Document.HighlightingDefinitionInvalidException: Could not load mode definition file '82252d18-7d24-4703-b2d3-24f538c9d532.xshd'.
 ---> System.MissingMethodException: Method 'System.Drawing.Color.black' not found.
   at System.RuntimeType.InvokeMember(String name, BindingFlags bindingFlags, Binder binder, Object target, Object[] providedArgs, ParameterModifier[] modifiers, CultureInfo culture, String[] namedParams)
   at ICSharpCode.TextEditor.Document.HighlightColor..ctor(XmlElement el)
   at ICSharpCode.TextEditor.Document.HighlightRuleSet..ctor(XmlElement el)
   at ICSharpCode.TextEditor.Document.HighlightingDefinitionParser.Parse(DefaultHighlightingStrategy highlighter, SyntaxMode syntaxMode, XmlReader xmlReader)
   --- End of inner exception stack trace ---
   at ICSharpCode.TextEditor.Document.HighlightingDefinitionParser.Parse(DefaultHighlightingStrategy highlighter, SyntaxMode syntaxMode, XmlReader xmlReader)
   at ICSharpCode.TextEditor.Document.HighlightingManager.LoadDefinition(DictionaryEntry entry)
   at ICSharpCode.TextEditor.Document.HighlightingManager.FindHighlighter(String name)
   at ICSharpCode.TextEditor.TextEditorControl.SetHighlighting(String name)
   at BasicPawn.ClassSyntaxTools.UpdateTextEditorSyntax()

```

I tried putting e

**Comment by Timocop** [comment]
Yea the log is clean. I'll add a "Debug-Mode" later to see whats going on.
Its probably a pathing issue with the Unix path.

I also tested BasicPawn with Mono in the current state and looks like its runs even worse then on Wine because the missing Windows API i use (Crypto, Process etc..).
I've tested it on Ubuntu x86 and Windows 8 x86 with mono installed. 
On Ubuntu it crashes instandly (probably because missing API or some thread crashes with an unhandled exception) and writes stuff into the application_error.log. 
On Windows 8 it runs fine, until the usage of unmanaged Windows API. I'll stick with Wine for a while and fix stuff there because its easier to fix.

**Comment by Timocop** [comment]
I could not reproduce the problem you're having. What Wine version are you using? I tested it with 1.8.x
Does it still use the Unix path instead of the Windows path? I dont get it why its using both paths...

**Comment by Keplyx** [comment]
> What Wine version are you using?

I'm using wine 2.6 (dev branch), I don't know how you are still in 1.8, [the stable branch is in 2.0.1](https://www.winehq.org/).

The problem may be that all my files (software, .sp, and includes) are in my home directory, not under wine virtual drive. I'm gonna try to move everything inside the .wine folder and I'll report back.

**Comment by Keplyx** [comment]
**Found the source of the problem** (I think).
I tried putting every files, including the software inside wine virtual drive, didn't change anything.
I tried creating an include folder in the same level as the secondary files (using the default profile), didn't change anything.
![spectacle m16612](https://cloud.githubusercontent.com/assets/23726131/25452582/6151cf30-2ac6-11e7-90a3-401eec6f1d67.png)

So I tried adding #includes in the .sp header, restarted the software, there it worked. Problem is, I put every #include in the main file, so I don't have to worry about duplicate #includes.
![spectacle j17115](https://cloud.githubusercontent.com/assets/23726131/25452630/899c3fd4-2ac6-11e7-8063-1cafeb2a60e2.png)

I think this is a case where the software uses auto complete based on the includes, but my secondary files don't have any.

**Comment by Timocop** [comment]
Tried to install .NET 3.5 on Wine 2.6 Dev but i could not get past .NET 2.0 SP1 (Winetricks hangs). Seems like 2.6 Dev is quite unstable for .NET programs or unstable in general idk. 2.0 Stable works fine tho. Any chance to test the stable one or is there a reason for you to use Dev version?

**Comment by Keplyx** [comment]
Well, I have .net 4.5 installed on wine 2.6. I'm using this version because I could not get past .net 2.0 with wine 2.0 for some reason

**Comment by Timocop** [comment]
Just to clearify, you're using [**winetricks**](https://wiki.winehq.org/Winetricks) to install .NET Framework right?

**Comment by Keplyx** [comment]
Yes, I installed .net with winetricks while under wine 2.4, I updated to 2.5 and then 2.6. I followed [those instructions](https://appdb.winehq.org/objectManager.php?sClass=version&iId=25478).

**Comment by Timocop** [comment]
Closing this for now, seems to be a Wine issue that im unable to reproduce.
I'll re-open this if other people have the same issue.

**Comment by Keplyx** [comment]
**Issue seems to be fixed with [v0.715](https://github.com/Timocop/BasicPawn/releases/tag/v0.715-beta):**

![spectacle lh4367](https://cloud.githubusercontent.com/assets/23726131/26397986/6042949a-4078-11e7-99fb-adc1c9da1fad.png)

By the way, the start page is really useful, nice job!

**Comment by Timocop** [comment]
Maybe your .NET 2.0 was broken because you mentioned that it didnt installed properly.
Changed target to .NET 3.5 now. Good to know that fixed it! :D
